corgi/core/migrations/0092_fix_nevra_formatting.py

```diff
@@ -0,0 +1,33 @@
+from django.db import migrations
+from django.db.models import F, Value, functions
+
+
+def fix_nevra_nvr_data(apps, schema_editor):
+    """Clean up components where the NEVRA or NVR ends with a dash, or a dash then dot"""
+    Component = apps.get_model("core", "Component")
+
+    # arch is not present in the NEVRA for this component type (whatever it may be)
+    Component.objects.filter(
+        nevra__endswith="-", epoch=0, version="", release="", arch="noarch"
+    ).update(nvr=F("name"), nevra=F("name"))
+
+    # arch is present in the NEVRA for this component type (whatever it may be)
+    Component.objects.filter(
+        nevra__endswith="-.noarch", epoch=0, version="", release="", arch="noarch"
+    ).update(nvr=F("name"), nevra=functions.Concat(F("name"), Value(".noarch")))
+    # TODO: Why does this query return results in stage?
```

Comment:
I think nevra only really useful for REDHAT RPMs or OCI types. For those Components we should append the .noarch suffix, but otherwise leave it off the nevra.